packages/esm-bed-management-app/src/ward-with-beds/ward-with-beds.component.tsx

```diff
@@ -87,10 +87,10 @@ const WardWithBeds: React.FC = () => {
     }));
   }, [paginatedData]);
 
-  const openNewBedModal = () => {
-    const dispose = showModal('new-bed-modal', {
-      closeModal: () => dispose(),
-      mutate: mutate,
+  const handleAddEditBedWorkspace = () => {
+    launchWorkspace('add-edit-bed-workspace', {
+      workspaceTitle: t('addBed', 'Add bed'),
+      mutateBeds: mutate,
       defaultLocation: { display: name, uuid: location },
     });
   };
```

Comment:
```suggestion
  type WorkspaceMode = 'add' | 'edit';
  
  const handleAddEditBedWorkspace = (mode: WorkspaceMode, bed?: Bed) => {
    launchWorkspace('add-edit-bed-workspace', {
      workspaceTitle: mode === 'add' ? t('addBed', 'Add bed') : t('editBed', 'Edit bed'),
      mutateBeds: mutate,
      defaultLocation: { display: name, uuid: location },
      ...(mode === 'edit' && bed ? { bed } : {}),
    });
  };
```

Should we change the workspace title based on whether we're creating or editing a bed?